### PR TITLE
[BUG] Constructor of any DL estimator to pass non-default values to underlying `Network` object

### LINKED ISSUE #4067: [BUG] Initializing any DL estimator with non-default values constructs these values only superficially
**Describe the bug**

A logical bug is present in all DL estimators and is caused by not passing custom parameters to its underlying `Network` since `Network` actually makes use of these values to create the DL model of appropriate size. Take a look at the example provided below.

**To Reproduce**

```python
from pprint import pprint
from classification.deep_learning import CNNClassifier

model = CNNClassifier(
    kernel_size=3,
    n_conv_layers=1,
)

# model.summary() fails due to self.history being None, another bug

pprint(vars(model))
print("=="*25)
pprint(vars(model._network))
```

Output:
``` python
{'_X_metadata': [],
 '_class_dictionary': {},
 '_estimator_type': 'classifier',
 '_is_fitted': False,
 '_network': CNNNetwork(),
 '_tags_dynamic': {},
 '_threads_to_use': 1,
 'activation': 'sigmoid',
 'avg_pool_size': 3,
 'batch_size': 16,
 'callbacks': None,
 'classes_': [],
 'fit_time_': 0,
 'history': None,
 'kernel_size': 3,
 'loss': 'mean_squared_error',
 'metrics': None,
 'model_': None,
 'n_classes_': 0,
 'n_conv_layers': 1,
 'n_epochs': 2000,
 'optimizer': None,
 'random_state': None,
 'use_bias': True,
 'verbose': False}
==================================================
{'_tags_dynamic': {},
 'activation': 'sigmoid',
 'avg_pool_size': 3,
 'filter_sizes': [6, 12],
 'kernel_size': 7,
 'n_conv_layers': 2,
 'random_state': 0}
```

**Expected behavior**

This is happening because of `self._network = CNNNetwork()` (or corresponding Network) in the constructor of the estimator. The values that we receive are to be passed into the Network.

**Additional context**

I'm speculating this is the reason why the checks at #4036 are failing despite having very small hyperparameters. In any case, I'll open a PR to resolve this bug.

**Versions**
<details>

```
D:\Anaconda\envs\sktime-dev\lib\site-packages\_distutils_hack\__init__.py:33: UserWarning: Setuptools is replacing distutils.
  warnings.warn("Setuptools is replacing distutils.")

System:
    python: 3.8.13 | packaged by conda-forge | (default, Mar 25 2022, 05:59:00) [MSC v.1929 64 bit (AMD64)]
executable: D:\Anaconda\envs\sktime-dev\python.exe
   machine: Windows-10-10.0.22621-SP0

Python dependencies:
          pip: 22.3.1
   setuptools: 65.5.1
      sklearn: 1.1.3
       sktime: 0.15.0
  statsmodels: 0.13.5
        numpy: 1.22.4
        scipy: 1.9.3
       pandas: 1.5.1
   matplotlib: 3.6.2
       joblib: 1.2.0
        numba: 0.56.4
     pmdarima: 2.0.1
      tsfresh: 0.19.0
```
</details>

### PR DESCRIPTION
**Pull request by achieveordie**
#### Reference Issues/PRs

Fixes #4067

#### What does this implement/fix? Explain your changes.

- [x] Manually correct behaviour for all current estimators by passing all values from estimator object to network object during estimator initialization.
- [x] Add a test to ensure this works properly.

#### Does your contribution introduce a new dependency? If yes, which one?

No

#### What should a reviewer concentrate their feedback on?

We need to discuss if there is another way that is superior to the current solution. 

##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/sktime/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/sktime/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

**Review by fkiraly — APPROVED** [review]
looks reasonable and like a good fix, but is still a draft - is this ready to be merged?

**Comment by achieveordie** [comment]
I've added a test to ensure it doesn't fail in the future. Should be ready to be merged now.

**Comment by fkiraly** [comment]
seems to have crashed while downloading `prophet`? I've restarted the failed CI now, I think it's queued.

**Comment by fkiraly** [comment]
failed again with timeout while downloading `prophet` from `pypi`? What is going on there?

**Comment by achieveordie** [comment]
I guess something was down temporarily, I saw the failures on other checks as well. I wonder why docs are not being built, any idea?

File: sktime/classification/deep_learning/tapnet.py [hunk 1]
```diff
@@ -142,7 +142,21 @@ def __init__(
...
+        self._network = TapNetNetwork(
+            dropout=self.dropout,
+            filter_sizes=self.filter_sizes,
+            kernel_size=self.kernel_size,
+            dilation=self.dilation,
+            layers=self.layers,
+            use_rp=self.use_rp,
+            rp_params=self.rp_params,
+            use_att=self.use_att,
+            use_ss=self.use_ss,
```
Comment (aaronrmm) [inline sktime/classification/deep_learning/tapnet.py:154]: There has no `self.use_ss`.

Comment (achieveordie) [reply sktime/classification/deep_learning/tapnet.py]: fixed, thanks!

**Review by fkiraly — APPROVED** [review]
Straightforward fix for a hard-to-detect and quite consequential bug.

Seems like all params are now passed on. We can deal with the general design problem later.

**Pull request merged**